text/0000-ember-cli-docs.md

```diff
@@ -0,0 +1,183 @@
+are currently a Middleman app. Similar migrations have been very successful.
+
+The rewrite and/or reorganization of content is driven by an audit of the existing
+content's relevance and balance. Following an audit fo the content, 
+it became clear that a greenfield approach is more time efficient and will lead to a 
+better experience for developers who are new to Ember.
+
+## Detailed design
+
+This app will have a new table of contents. The architecture will follow the
+same patterns successfully used in other apps that have been converted from
+Middleman apps to Ember.
+
+### User Personas
+
+The content layout should follow the progression of an Ember developer's
+learning experience. We have identified up to four user personas for the 
+CLI documentation:
+
+1. A new or "typical" Ember CLI user - someone whose primary work is
+running common commands like `ember serve` and who has a "zero
+config" type of experience with Ember
+2. Power users - developers who make their own configurations to the
+build pipeline
+3. Beginner addon authors - those who are looking to build simple
+shared UI components, methods, or wrappers for existing npm libraries
+4. Advanced addon authors - those who dig into internals to make their
+addon work, or who are planning for broad extensibility
+
+### Table of Contents
+
+Applying these User Personas to the CLI content, the following Table of
+Contents emerges. "Beginner" topics will include links to later "Advanced"
+topics, similar to how the Guides link to the API docs. This topic 
+arrangement may change as writing progresses, but community feedback is
+requested so no important topics are skipped.
+
+- Landing page 
+    - how to install ember cli 
+    - a very simple, short definition of what it is (the official way to create, build, and test an Ember app)
+- Overview
+    - What is the CLI
+    - Docs layout
+    - Why it should be used
+    - How to contributoe
+- Quickstart Tutorial
+    - ember new
+    - ember server
+    - ember test
+    - ember generate
+    - ember install (incl some general info about ember addons vs npm packages)
+- Basic use (explain options of each)
+    - using the "help" commmand
+    - ember new
+    - ember server
+    - ember test
+    - ember generate
+    - ember install (incl link to later section on shims for npm packages)
+    - feature flags & configurations
+    - Environmental variables
+    - File tree reference
+    - addons/dependencies
+    - Upgrading
+- Advanced use
+    - shims
+    - broccoli
+    - custom blueprints
+    - CSS compilation
+    - Using another testing library
+    - more on dependencies
+    - more configurations
+- Addon authoring
+    - Overview
+    - What is an addon (technically)
+    - Learning to build addons
+    - Creating a standalone addon
+    - Creating an in-repo addon
+    - Making an npm package wrapper
+    - Using the dummy app
+    - Including assets
+    - Testing your addon
```

Comment:
This one is a big topic for me. Writing ember-qunit tests using the dummy app is fairly straightforward, but writing node tests (with mocha, jest, qunit-cli, etc) especially for broccoli-related code is under-specified.